services/src/handlers/workflows.rs

```diff
@@ -1450,4 +1510,144 @@ mod tests {
             result.as_slice()
         );
     }
+
+    #[cfg(feature = "xgboost")]
+    #[tokio::test]
+    #[serial]
+    async fn ml_model_from_workflow_task_success() {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>().unwrap();
+        let cfg_backup = cfg.model_defs_path;
+
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let tmp_path = tmp_dir.path();
+        std::fs::create_dir_all(tmp_path.join("pro/ml/xgboost")).unwrap();
+
+        let temp_ml_path = tmp_path.join("pro/ml").to_str().unwrap().to_string();
+
+        set_config("machinelearning.model_defs_path", temp_ml_path).unwrap();
```

Comment:
why do you need to change this path? Can't you set a path for testing in the Settings-test.toml?